test/karma.setup.js

```diff
@@ -0,0 +1,58 @@
+// // the jest.fn() API
+// import jest from "jest-mock";
+// // The matchers API
+// import expect from "expect";
+
+// Add missing Jest functions
+// window.test = window.it;
+// window.test.each = (inputs) => (testName, test) =>
+//   inputs.forEach((args) => window.it(testName, () => test(...args)));
+// window.test.todo = function () {
+//   return undefined;
+// };
+// window.jest = jest;
+// window.expect = expect;
```

Comment:
this part is jest browser polyfills as suggested [here](https://github.com/tom-sherman/blog/blob/main/posts/02-running-jest-tests-in-a-browser.md), needs bundling so will wait